src/main/flight/pid.c

```diff
@@ -292,13 +315,99 @@ static void pidApplyRateController(const pidProfile_t *pidProfile, pidState_t *p
 #endif
 }
 
+void updateMagHoldHeading(int16_t heading)
+{
+    magHoldTargetHeading = heading;
+}
+
+int16_t getMagHoldHeading() {
+    return magHoldTargetHeading;
+}
+
+uint8_t getMagHoldState()
```

Comment:
I see wou wrapped the MAG_HOLD selection logic into a separate function - good idea, it never came to me.